plugins/module_utils/mso.py

```diff
@@ -1652,10 +1664,11 @@ def nd_request(self, path, method=None, data=None, file=None, qs=None, prefix=""
                     self.fail_json(msg="ND Error: Unknown error no error code in decoded payload".format(**payload), data=data, info=info, payload=payload)
             else:
                 self.result["raw"] = info.get("raw")
-                # Connection error
-                msg = "Connection failed for {0}. {1}".format(info.get("url"), info.get("msg"))
-                self.error = msg
-                self.fail_json(msg=msg)
+                if not ignore_not_found_error:
```

Comment:
why set the raw result when the error might be ignore? should `elif not ignore_not_found_error` not keep current behaviour?

Reply:
Done!